### PR TITLE
PyUtils Cleanup

### PR DESCRIPTION
**Pull request by fthaler**
- Much simplified SLURM execution (hopefully with similar stability).
- Automatic SLURM detection.
- Better error catching in environment loading.

### CODE REVIEW

**Comment by fthaler** [comment]
launch jenkins

**Comment by fthaler** [comment]
> Why is slurm so much simpler?

Because we don’t need to spawn dozens of perftest executions anymore. Should be enough to run a single command now in all cases (perftests are not yet included).

> Do we need to change jenkins to try it?

Tests should work, perftests not yet.

**Comment by fthaler** [comment]
launch jenkins

**Comment by fthaler** [comment]
launch jenkins

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -30,14 +30,13 @@ jobs:
     - name: setup environment
       run: |
         export num_proc=$(nproc)
-        echo "::set-env name=GTRUN_NO_SLURM::true"
```
Comment (havogt) [inline .github/workflows/tests.yml:33]: please change line 10 above to something that fits better, e.g. just `tests` instead of `perftests`

**Comment by havogt** [comment]
The CUDA build takes 26 minutes on gh actions. I think that's too much. I'd prefer having the option to run a smaller test.

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**